### ISSUE TITLE
Token advisory

### ISSUE THREAD

**Issue by iProdigy** [TC-C/vod_search#2]
Just a reminder - try to avoid committing oauth tokens publicly (especially first-party tokens). I revoked the ones in https://github.com/TC-C/vod_search/blob/654fc285eeb5dc965197e1afd6378ca2826e8073/src/twitch/twitch_reader.rs

Have a good day!

**Comment by pravinxor** [comment]
Ty for letting me know. I don’t have the resources to leave a server running for handling requests, so I thought it would be easier to leave them in the program. Is it okay to hard-code the ones found (publicly) when reading twitch’s network requests?

**Comment by iProdigy** [comment]
You can hardcode whatever tokens, but they shouldn't be shared publicly. For example, sharing the first-party token that is used when you're logged into twitch would allow other folks to have a great deal of control over your account

I just revoked the latest one in https://github.com/TC-C/vod_search/commit/8fd2bba3620e58b11f8fa6e348c9a0df14bd9807

That token provides these permissions:
```json
{"client_id":"kimne78kx3ncx6brgo4mv6wki5h1ko","login":"ohsnapitsnathn","scopes":["chat_login","user_presence_friends_read","user_read","user_subscriptions","channel_commercial","channel_editor","channel_feed_edit","channel_feed_read","channel_feed_report","channel_read","collections_edit","communities_edit","communities_moderate","user_edit","user_entitlements_read","user_follows_edit","user_friends_edit","user_friends_read","user_presence_edit","user_subscriptions_edit"],"user_id":"196109595","expires_in":0}
```

so it's really not a good idea to share it

**Comment by iProdigy** [comment]
Also it seems like you are sharing your client secret for `cuwhphy3xzy01xn60rddmr57x8hzc6` - this is also a bad idea & should not be posted publicly. You can go to the twitch dev console to generate a new client secret.

**Comment by pravinxor** [comment]
I will definitely keep this in mind. Thank you very much for letting me know and taking action before anything bad happened. I am a novice to this and hadn't considered the level of access that the oauth token twitch assigns has.

**Comment by iProdigy** [comment]
In https://github.com/TC-C/vod_search/commit/564a1677f8a5f938d91caaf80e9e47bd08258e9b you (accidentally?) committed your first-party oauth token again - I've revoked it for you

Perhaps you should add a `.env` file that is gitignore'd & contains your token so you can read from that file instead of hardcoding it :) https://crates.io/crates/dotenv

**Comment by pravinxor** [comment]
Yep that was an accident. I was testing to see if my stuff still worked after some changes I made and forgot to remove my token later on. I've moved everything over to graphql for the sake of not having to deal with the frustration that comes with using helix (unnecessarily needing private information for simple api calls) at the expense of some stability and support.

Thank you very much for your help!
